### DISCUSSION TITLE
Add option to clear query string when component is unmounted

### DISCUSSION THREAD

**Discussion by atahanyorganci** [47ng/nuqs#663] [Ideas] [answered]
Clearing when the query string when the component that uses the state can help with keeping the URL clean. This is extra important for caching behavior of Next.js. 

For example if a user navigates to `/sale` from `/browse` clicks a button to set `size` state (assume URL is `/sale?size=large` when they navigate back to  with `/browse?size=large` the caching behavior of Next.js will invalidate the client cache.

**Comment by atahanyorganci** [comment, marked as answer]
This is easliy fixed on application code by `useEffect` clean up function.

**Reply by franky47** [reply]
I'm going to mark this as the answer, as I believe this behaviour should belong in userland.

The main reason is that there can be several hooks mounted on the same key, and one of them unmounting but not the others could lead to unexpected behaviours.

**Reply by atahanyorganci** [reply]
For clarity, ysing `useQueryState` with same key isn't an antipattern then.

**Reply by franky47** [reply]
Not an antipattern at all!

All hooks using a given key are synced together, so you could have a control component that only gets the state updater function for writes, and another part of the UI far away in the tree doing just reads for display, like you would with a global state manager like Zustand or Redux (except the storage is the URL, and the selector is the search param name).

**Reply by atahanyorganci** [reply]
Thanks! Very helpful
